In [45]:
import pandas as pd
data = pd.read_csv('GDPlist.csv', encoding = 'unicode_escape')
data.head(5)

Country      Continent  GDP (millions of US$)
0   Afghanistan           Asia                  18181
1       Albania         Europe                  12847
2       Algeria         Africa                 190709
3        Angola         Africa                 100948
4     Argentina  South America                 447644

In [46]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 125 entries, 0 to 124
Data columns (total 3 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Country                125 non-null    object
 1   Continent              125 non-null    object
 2   GDP (millions of US$)  125 non-null    int64 
dtypes: int64(1), object(2)
memory usage: 3.1+ KB


In [47]:
data.shape

(125, 3)

Dữ liệu có 125 hàng, 3 cột, không bị khuyết thiếu

In [48]:
# Việt hóa tên các cột trong bảng dữ liệu: Country 🡪 Nuoc; Continent 🡪 Chauluc; GDP (millions of US$) 🡪 GDP (trieu $)
data.rename(columns={'Country':'Nuoc','Continent':'Chauluc','GDP (millions of US$)':'GDP (trieu $)'}, inplace=True)
data.head(2)

Nuoc Chauluc  GDP (trieu $)
0   Afghanistan    Asia          18181
1       Albania  Europe          12847

In [49]:
# Chèn thêm một cột “Thanhpho” vào sau cột “Nuoc”, giá trị ban đầu là giá trị của cột “Nuoc” 
data.insert(1, 'Thanhpho', data['Nuoc'],True)
data.head(2)

Nuoc      Thanhpho Chauluc  GDP (trieu $)
0   Afghanistan   Afghanistan    Asia          18181
1       Albania       Albania  Europe          12847

In [64]:
# Trong cột Thanhpho, thay giá trị Vietnam thành Hanoi; Làm tương tự với các nước còn lại.
data['Thanhpho']=data['Thanhpho'].str.replace('Vietnam', 'Hanoi')
data

Nuoc      Thanhpho        Chauluc  GDP (trieu $)
0     Afghanistan   Afghanistan           Asia          18181
1         Albania       Albania         Europe          12847
2         Algeria       Algeria         Africa         190709
3          Angola        Angola         Africa         100948
4       Argentina     Argentina  South America         447644
..            ...           ...            ...            ...
120    Uzbekistan    Uzbekistan           Asia          45353
121     Venezuela     Venezuela  South America         315841
122       Vietnam         Hanoi           Asia         122722
123         Yemen         Yemen         Africa          33675
124        Zambia        Zambia         Africa          19206

[125 rows x 4 columns]

In [68]:
# Xóa các bản ghi có Chauluc là ‘Asia’
# Xóa các bản ghi có GDP < 300000
data.drop(data[data['Chauluc']=='Asia'].index, inplace=True)
data.drop(data[data['GDP (trieu $)']<300000].index, inplace=True)

### Thực hành phân tích các giá trị thống kê của biến định tính, định lượng; Pivot table, Hợp nhất dữ liệu. 

In [69]:
import pandas as pd
data = pd.read_csv('GDPlist.csv', encoding = 'unicode_escape')
data.head(5)

Country      Continent  GDP (millions of US$)
0   Afghanistan           Asia                  18181
1       Albania         Europe                  12847
2       Algeria         Africa                 190709
3        Angola         Africa                 100948
4     Argentina  South America                 447644

###### Tính giá trị lớn nhất và nhỏ nhất của GDP.

In [74]:
print('Giá trị nhỏ nhất của GDP: ', data.iloc[:,2].min())
print('Giá trị lớn nhất của GDP: ', data.iloc[:,2].max())

Giá trị nhỏ nhất của GDP:  10600
Giá trị lớn nhất của GDP:  15094025


###### Hãy cho biết xu hướng phân bố dữ liệu của GDP.

Biến dữ liệu định tính

In [100]:
data_country = data.loc[:,'Continent']
data_country.mode()

0    Europe
Name: Continent, dtype: object

The continent has the most numbers of contries is Europe

In [99]:
data_country.value_counts()

Europe           37
Asia             36
Africa           29
North America    11
South America     9
Oceania           3
Name: Continent, dtype: int64

Biến dữ liệu định lượng

In [90]:
data.describe()

GDP (millions of US$)
count           1.250000e+02
mean            5.554278e+05
std             1.649266e+06
min             1.060000e+04
25%             2.574200e+04
50%             9.608900e+04
75%             3.332380e+05
max             1.509402e+07

In [93]:
data['GDP (millions of US$)'].median()

96089.0

In [94]:
data['GDP (millions of US$)'].mean()

555427.784

=> Lệch trái

###### Với mỗi châu lục hãy tính tổng GDP; trung bình cộng GDP. Hợp nhất 2 bảng này thành một bảng duy nhất gồm 3 thông tin: Tên châu lục; Tổng GDP; TBC GDP.

In [115]:
table_1 = data.pivot_table(values='GDP (millions of US$)', index='Continent', aggfunc='sum')
table_1.rename(columns={'GDP (millions of US$)': 'Total of GDP'},inplace= True)
table_2 = data.pivot_table(values='GDP (millions of US$)', index='Continent', aggfunc='mean')
table_2.rename(columns={'GDP (millions of US$)': 'Average of GDP'},inplace= True)
table_new = pd.concat([table_1, table_2], axis=1)
table_new

Total of GDP  Average of GDP
Continent                                  
Africa              1820811    6.278659e+04
Asia               24425891    6.784970e+05
Europe             19139380    5.172805e+05
North America      18238447    1.658041e+06
Oceania             1662727    5.542423e+05
South America       4141217    4.601352e+05